assets/src/scripts/selectors/timeseriesSelector.js

```diff
@@ -0,0 +1,75 @@
+const memoize = require('fast-memoize');
+const { createSelector } = require('reselect');
+
+/*
+ * Selectors that return properties from the state
+ */
+export const getVariables = state => state.series.variables ? state.series.variables : null;
+
+export const getCurrentVariableID = state => state.timeseriesState.currentVariableID;
+
+export const getCurrentDateRange = state => state.timeseriesState.currentDateRange;
+
+export const getMethods = state => state.series.methods ? state.series.methods : null;
+
+export const getQueryInfo = state => state.series.queryInfo || {};
+
+/*
+ * Selectors the return derived data from the state
+ */
+
+/*
+ * @return {Object}     Variable details for the currently selected variable or null.
+ */
+export const getCurrentVariable = createSelector(
+    getVariables,
+    getCurrentVariableID,
+    (variables, variableID) => {
+        return variableID  && variables  && variables[variableID] ? variables[variableID] : null;
+    }
+);
+
+/*
+ * @return {String} or null - The parameter code of the currently selected variable or null if no variable selected
+ */
+export const getCurrentParmCd = createSelector(
+    getCurrentVariable,
+    (currentVar) => {
+        return currentVar && currentVar.variableCode ? currentVar.variableCode.value : null;
+    }
+);
+
+const tsRequestKey = function(tsKey, period, parmCd) {
+    return tsKey === 'median' || period === 'P7D' ? tsKey : `${tsKey}:${period}:${parmCd}`;
+};
```

Comment:
I'm a little confused by what's going on here - could we put P7D in the key for consistency?

Reply:
First question's answer: I don't think so. The issue is that sometimes we use tsKey as one "current", "compare", or "median". The request key is a little different as it's the key used in the requests. These end up being "current", "compare", "median", and then keys with whatever extended data we have retrieved.

second question's answer: That is what I would expect to see in series.timeSeries

Reply:
Oh the request key, gotcha

Reply:
So re the first question: I will change the key to use the P7D except for the median. I'm going to wait and see what Andrew comes up with before changing anything about median data.

Reply:
Actually I'm going to wait on this. It is turning into a bigger deal and I'd like to get this code out.

Reply:
So, I'm still working on the rejiggering the median data, here's what I I've been scheming so far: https://github.com/ayan-usgs/waterdataui/blob/wdfn-221-median-data-step-function/assets/src/scripts/components/hydrograph/statistics.js#L10

That function is intended to be able to figure out what dates to display based on the date range of the timeseries. Hopefully, that'll allow work to continue while I get the median step function sorted out.